lib/diff.mli

```diff
@@ -1,22 +1,28 @@
+open Types
```

Comment:
I am not sure I like the global open, as we define here types of very similar name that what can be found in `Types`: for instance, `sig_item` comes from this file, but `signature` comes from `Types`. Similarly for `module`, `values`, etc.
It makes it harder to read, and raise the chance of an accidental shadowing.
Could you revert this open?